docs/developer-guide/debugging.md

```diff
@@ -1,26 +1,130 @@
+
+Steps for configuring this extension are described next.
 
-## Install IDE
+### Download SVD file
 
-### QtCreator
+A System View Description (SVD) file describes the configuration (CPU, peripherals, registers, etc.) of the microcontroller. The Cortex-Debug extension can make use of an SVD file to provide more detailed debugging information, such as the ability to inspect register values.
+
+SVD files can be downloaded from STMicroelectronics. The files for the F4 are contained in the ZIP file that can be downloaded [here](https://www.st.com/resource/en/svd/stm32f4_svd.zip), and the relevant file is `STM32F405.svd`. The files for the F1 are contained in the ZIP file that can be downloaded [here](https://www.st.com/resource/en/svd/stm32f1_svd.zip), and the relevant file is `STM32F103.svd`. Put those files in a convenient location.
+
+### Configure build step
+
+You can configure VS Code to run `make` for you when you press `Ctrl+Shift+B`. To do this, put the following in `.vscode/tasks.json` inside your firmware working directory:
+
+``` json
+{
+  // See https://go.microsoft.com/fwlink/?LinkId=733558
+  // for the documentation about the tasks.json format
+  "version": "2.0.0",
+  "tasks": [
+    {
+      "type": "shell",
+      "label": "make",
+      "command": "make",
+      "args": ["DEBUG=GDB"],
+      "group": {
+        "kind": "build",
+        "isDefault": true
+      }
+    }
+  ]
+}
+```
+
+Note that by default, this will only build the F4 (Revo) firmware. To build the F1 firmware, you will need to edit this to add the argument `BOARD=NAZE`.
+
+### Configure debugging
+
+To configure in-circuit debugging of F4 and F1 targets, put something like the following in `.vscode/launch.json` inside your firmware working repository:
+
+``` json
+{
+  // Use IntelliSense to learn about possible attributes.
+  // Hover to view descriptions of existing attributes.
+  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+  "version": "0.2.0",
+  "configurations": [
+    {
+      "name": "STM32F405",
+      "type": "cortex-debug",
+      "request": "launch",
+      "servertype": "openocd",
+      "cwd": "${workspaceRoot}",
+      "executable": "${workspaceRoot}/boards/airbourne/build/rosflight_REVO_Debug.elf",
+      "device": "STM32F405",
+      "svdFile": "/home/dpkoch/Documents/SVD/STM32F405.svd",
+      "configFiles": [
+        "interface/stlink-v2.cfg",
+        "target/stm32f4x.cfg"
+      ],
+      "runToMain": true
+    },
+    {
+      "name": "STM32F103",
+      "type": "cortex-debug",
+      "request": "launch",
+      "servertype": "openocd",
+      "cwd": "${workspaceRoot}",
+      "executable": "${workspaceRoot}/boards/breezy/build/rosflight_NAZE_Debug.elf",
+      "device": "STM32F103",
+      "svdFile": "/home/dpkoch/Documents/SVD/STM32F103.svd",
```

Comment:
maybe `/path/to/STM32F103.svd`?
(otherwise, people will get a weird error like 'no such path `/home/dpkoch`.. which can be confusing to newbies)

Reply:
Oh yeah, that's what I intended to do but apparently forgot to change it. Good catch.